tests/test_file_in_out.py

```diff
@@ -31,24 +32,29 @@
 
 
 @pytest.mark.slow
-@pytest.mark.parametrize("flush", [False, True])
+@pytest.mark.parametrize("input_type", ["csv", "jsonlines", "parquet"])
 @pytest.mark.parametrize("output_type", ["csv", "json", "jsonlines"])
+@pytest.mark.parametrize("flush", [False, True])
 @pytest.mark.parametrize("repeat", [1, 2, 5])
-def test_file_rw_pipe(tmp_path, config, output_type, flush, repeat: int):
-    input_file = os.path.join(TEST_DIRS.tests_data_dir, "filter_probs.csv")
+def test_file_rw_pipe(tmp_path, config, input_type, output_type, flush, repeat: int):
+    input_file = os.path.join(TEST_DIRS.tests_data_dir, 'filter_probs.{}'.format(input_type))
+    validation_file = os.path.join(TEST_DIRS.tests_data_dir, "filter_probs.csv")
     out_file = os.path.join(tmp_path, 'results.{}'.format(output_type))
 
+    if (input_type == "parquet"):
+        CppConfig.set_should_use_cpp(False)
+
```

Comment:
@mdemoret-nv get a segfault in file source stage using parquet when cpp is enabled

Reply:
Its because you didnt make a C++ implementation of the serializer/deserializer so it falls back to CSV. I've made some updates to add this for you